NineChronicles.Headless.AccessControlCenter/Controllers/AccessControlServiceController.cs

```diff
@@ -39,10 +40,17 @@ public ActionResult AllowAccess(string address)
         [HttpGet("entries")]
         public ActionResult<List<string>> ListBlockedAddresses(int offset, int limit)
         {
-            return _accessControlService
-                .ListBlockedAddresses(offset, limit)
-                .Select(a => a.ToString())
-                .ToList();
+            try
+            {
+                return _accessControlService
+                    .ListBlockedAddresses(offset, limit)
+                    .Select(a => a.ToString())
+                    .ToList();
+            }
+            catch (ArgumentException ex)
+            {
+                return BadRequest(ex.Message);
```

Comment:
It seems to be `InternalServerError` instead of `BadRequest` or I think the `AccessConrolServiceController.ListBlockedAddresses(...)` should check the `limit` variable manually.

Reply:
Both ACS service have different  limit values.
Is this a proper way to implement limit validation?
```csharp
AccessControlServiceController.cs > ListBlockedAddresses
...
            var maxLimit = 10;
            if (_accessControlService.GetType() == typeof(MutableRedisAccessControlService))
            {
                maxLimit = 10;
            }
            else if (_accessControlService.GetType() == typeof(MutableSqliteAccessControlService))
            {
                maxLimit = 100;
            }

            if (limit > maxLimit)
            {
                throw new ArgumentException($"The limit cannot exceed {maxLimit}.", nameof(limit));
            }
```

Reply:
The reason I thought it was necessary to validate directly at the controller level is because `ArgumentException` is such a common error. It can also happen in the constructor of `Address` if Redis has an unintended invalid key in it, which is usually not the case. One way to do this would be to **(1) create a separate, dedicated exception that inherits from `ArgumentException`.** Or, **(2) provide a value for Limit as a read-only property in the `IMutableAccessControlService` interface, and use it.** Or, **(3) go with something like the code you posted, and use the `is` keyword to check for type.**

Reply:
How about this?

```
        [HttpGet("entries")]
        public ActionResult<List<string>> ListBlockedAddresses(int offset, int limit)
        {
            var maxLimit = 10;
            if (_accessControlService is MutableRedisAccessControlService)
            {
                maxLimit = 10;
            }
            else if (_accessControlService is MutableSqliteAccessControlService)
            {
                maxLimit = 100;
            }

            if (limit > maxLimit)
            {
                return BadRequest($"The limit cannot exceed {maxLimit}.");
            }

            return _accessControlService
                .ListBlockedAddresses(offset, limit)
                .Select(a => a.ToString())
                .ToList();
        }
```
implement option 3

Reply:
LGTM 👍🏻